### PR TITLE
Update fingerprinting code to use the SQLite API

### PR DESCRIPTION
**Pull request by mistachkin**

### CODE REVIEW

**Comment by nirbhayc** [comment]
If it's meant to test the prepare phase, I'd suggest adding it as a knob in clnt (SET) so that it won't affect other connections.

File: db/db_fingerprint.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            t->count++;
+            t->cost += cost;
+            t->time += time;
+            t->rows += nrows;
+            assert( memcmp(t->fingerprint,fingerprint,FINGERPRINTSZ)==0 );
+            assert( t->zNormSql!=zNormSql );
+            assert( t->nNormSql==nNormSql );
+            assert( strncmp(t->zNormSql,zNormSql,t->nNormSql)==0 );
+            free(zNormSql);
+        }
```
Comment (riverszhang89) [inline db/db_fingerprint.c]: You need to call `reqlog_set_fingerprint(logger, fingerprint);` at the end. Otherwise reqlog will not have the fingerprint.

**Comment by mponomar** [comment]
Err, sorry.  I was commenting on the last commit that changes prepare behavior, not the entire PR.

**Comment by mistachkin** [comment]
Things are working much better now.  Would appreciate some review if/when you have time.

**Comment by mistachkin** [comment]
This pull request is now fully passing both test suites.

**Pull request merged**